Python/import.c

```diff
@@ -1369,11 +1411,21 @@ import_find_extension(PyThreadState *tstate,
         }
         struct _Py_ext_module_loader_result res;
         if (_PyImport_RunModInitFunc(def->m_base.m_init, info, &res) < 0) {
+            _Py_ext_module_loader_result_apply_error(&res, name_buf);
+            _Py_ext_module_loader_result_clear(&res);
             return NULL;
         }
-        assert(!PyErr_Occurred());
+        assert(!PyErr_Occurred() && res.err == NULL);
```

Comment:
Let this tell you which one went wrong.
```suggestion
        assert(!PyErr_Occurred());
        assert(res.err == NULL);
```

Reply:
done